### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by yaaszp**

### CODE REVIEW

File: network.tf [hunk 1]
```diff
@@ -0,0 +1,35 @@
+resource "azurerm_resource_group" "rg" {
+  name     = "${var.prefix}-rg"
+  location = "West Europe"
+}
+
+resource "azurerm_virtual_network" "vnet" {
+  name                = "${var.prefix}-virtual-network"
+  address_space       = ["10.0.0.0/16"]
```
Comment (NazarSheremet) [inline network.tf:8]: It would be better to move this value to variables

File: network.tf [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  location            = azurerm_resource_group.rg.location
+  resource_group_name = azurerm_resource_group.rg.name
+}
+
+resource "azurerm_subnet" "subnet" {
+  name                 = "internal"
+  resource_group_name  = azurerm_resource_group.rg.name
+  virtual_network_name = azurerm_virtual_network.vnet.name
+  address_prefixes     = ["10.0.2.0/24"]
```
Comment (NazarSheremet) [inline network.tf:17]: It would be better to move this value to variables

File: network.tf [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+locals {
+  nic_name = ["1", "2"]
+}
```
Comment (NazarSheremet) [inline network.tf:22]: It would be better to create a separate `locals.tf` file and move this value into it.

File: network.tf [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+resource "azurerm_network_interface" "nic" {
+  count               = length(local.nic_name)
```
Comment (NazarSheremet) [inline network.tf:25]: It's the single place where this value is used as I see. So will be better to use integer variables

File: network.tf [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  name                = "${var.prefix}-nic-${count.index + 1}"
+  location            = azurerm_resource_group.rg.location
+  resource_group_name = azurerm_resource_group.rg.name
+
+  ip_configuration {
+    name                          = "ipconfig1"
+    subnet_id                     = azurerm_subnet.subnet.id
+    private_ip_address_allocation = "Dynamic"
+  }
+}
```
Comment (NazarSheremet) [inline network.tf:35]: image

Add EoF

**Review by NazarSheremet — CHANGES_REQUESTED** [review]

File: network.tf (hunk 1 above)
Comment (yaaszp) [reply network.tf:8]: fixed

File: network.tf (hunk 2 above)
Comment (yaaszp) [reply network.tf:17]: fixed

File: network.tf (hunk 3 above)
Comment (yaaszp) [reply network.tf:22]: fixed

File: network.tf (hunk 4 above)
Comment (yaaszp) [reply network.tf:25]: fixed

**Review by NazarSheremet — CHANGES_REQUESTED** [review]
Check if you push your changes, because I don't see any commit after my review

**Comment by yaaszp** [comment]
> Check if you push your changes, because I don't see any commit after my review

Honestly, i don't understand what do you mean

**Comment by yaaszp** [comment]
> Check if you push your changes, because I don't see any commit after my review

Снимок экрана 2024-07-22 174022
Снимок экрана 2024-07-22 174449

File: network.tf (hunk 4 above)
Comment (YuriiSmolii) [reply network.tf:21]: Not fixed.

File: outputs.tf [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+  value = [for vm in azurerm_virtual_machine.vm : upper(vm.name)]
+}
+
+output "tags_string" {
+  value = join(", ", [for k, v in azurerm_virtual_machine.vm[0].tags : "${k}=${v}"])
+}
+
+output "vm_ids" {
+  value = [for vm in azurerm_virtual_machine.vm : vm.id]
+}
```
Comment (YuriiSmolii) [inline outputs.tf:11]: EOF

File: provider.tf [hunk 7]
```diff
@@ -0,0 +1,3 @@
+provider "azurerm" {
+  features {}
+}
```
Comment (YuriiSmolii) [inline provider.tf:3]: EOF

File: variables.tf [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+      direction                  = "Outbound"
+      access                     = "Deny"
+      protocol                   = "*"
+      source_port_range          = "*"
+      destination_port_range     = "*"
+      source_address_prefix      = "*"
+      destination_address_prefix = "*"
+    }
+  ]
+}
```
Comment (YuriiSmolii) [inline variables.tf:65]: EOF

File: vmss.tf [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+      direction                  = security_rule.value.direction
+      access                     = security_rule.value.access
+      protocol                   = security_rule.value.protocol
+      source_port_range          = security_rule.value.source_port_range
+      destination_port_range     = security_rule.value.destination_port_range
+      source_address_prefix      = security_rule.value.source_address_prefix
+      destination_address_prefix = security_rule.value.destination_address_prefix
+    }
+  }
+}
```
Comment (YuriiSmolii) [inline vmss.tf:57]: EOF

**Review by YuriiSmolii — CHANGES_REQUESTED** [review]
Changes requested.

File: network.tf (hunk 4 above)
Comment (yaaszp) [reply network.tf:21]: if I understood you correct, I don't need to use local value here.
Fixed

File: outputs.tf (hunk 6 above)
Comment (yaaszp) [reply outputs.tf:11]: fixed

File: provider.tf (hunk 7 above)
Comment (yaaszp) [reply provider.tf:3]: fixed

File: vmss.tf (hunk 9 above)
Comment (yaaszp) [reply vmss.tf:57]: fixed

File: variables.tf (hunk 8 above)
Comment (yaaszp) [reply variables.tf:65]: fixed

**Comment by NazarSheremet** [comment]
> > Check if you push your changes, because I don't see any commit after my review
> 
> Honestly, i don't understand what do you mean

Hmmm, interesting. I don't see that commit when the previous time review your solution

**Review by NazarSheremet — APPROVED** [review]
Good job!

**Pull request closed without merging**